validator_client/src/lib.rs

```diff
@@ -142,15 +145,63 @@ impl<T: EthSpec> ProductionValidatorClient<T> {
             );
         }
 
+        let beacon_node_urls: Vec<Url> = config
+            .beacon_nodes
+            .iter()
+            .map(|s| s.parse())
+            .collect::<Result<_, _>>()
+            .map_err(|e| format!("Unable to parse beacon node URL: {:?}", e))?;
+        let beacon_nodes: Vec<BeaconNodeHttpClient> = beacon_node_urls
+            .into_iter()
+            .map(|url| {
+                let beacon_node_http_client = ClientBuilder::new()
+                    .timeout(HTTP_TIMEOUT)
+                    .build()
+                    .map_err(|e| format!("Unable to build HTTP client: {:?}", e))?;
+                Ok(BeaconNodeHttpClient::from_components(
+                    url,
+                    beacon_node_http_client,
+                ))
+            })
+            .collect::<Result<Vec<BeaconNodeHttpClient>, String>>()?;
+
+        let candidates = beacon_nodes
+            .into_iter()
+            .map(CandidateBeaconNode::new)
+            .collect();
+        let mut beacon_nodes: BeaconNodeFallback<_, T> =
+            BeaconNodeFallback::new(candidates, context.eth2_config.spec.clone(), log.clone());
+
+        // Perform some potentially long-running initialization tasks.
+        let (genesis_time, genesis_validators_root, fork) = tokio::select! {
+            tuple = init_from_beacon_node(&beacon_nodes, &context) => tuple?,
```

Comment:
If we init from the beacon node *before* we initialize validators then we have this issue where we don't start decoding keypairs until after genesis.

This has burned me with some testnets where we have lots of validators, we miss the first few epochs cause we're decrypting keypairs.

I'm running out of time to figure out a solution right now, but perhaps we can find something?

Reply:
Oh I didn't realize this would happen.. I've updated it so that we don't enable doppelganger detection here. Since refactoring doppelganger detection into a separate service we don't need to anyways because when the service starts up (post-genesis) it'll always update doppelganger detection periods